snuba/state/cache/redis/scripts/get.lua

```diff
@@ -1,25 +1,15 @@
--- KEYS[1]: The value key.
--- KEYS[2]: The waiting queue key.
--- KEYS[3]: The task unique ID key.
+-- KEYS[1]: The waiting queue key.
+-- KEYS[2]: The task unique ID key.
 -- ARGV[1]: The task execution timeout. Only used when creating a new task.
 -- ARGV[2]: The task unique ID. Only used when creating a new task.
-local value_key = KEYS[1]
-local wait_queue_key = KEYS[2]
-local task_id_key = KEYS[3]
+local wait_queue_key = KEYS[1]
+local task_id_key = KEYS[2]
 local task_timeout = ARGV[1]
 local task_id = ARGV[2]
 
-local CODE_RESULT_VALUE = 0
 local CODE_RESULT_EXECUTE = 1
 local CODE_RESULT_WAIT = 2
 
--- Check to see if a value already exists at the result key. If one does, we
```

Comment:
I'm not sure I understand why removing this is necessary.

Reply:
Previously, `execute_with_readthrough_caching` was responsible for:
- run get.lua which does:
  - check cache 
  - run value generation if cache miss
  - store cache

Now that we do the cache check in the very beginning in `DBQuery.db_query()`, we only get to executing this script if the value did not exist already. So the "readthrough" pattern is on the DBQuery class itself instead of within this script.
New order of ops is:

db_query():
- check cache

if cache miss:

_execute_and_cache_query():
- run get.lua which does:
  - run value generation
  - store cache

Reply:
`get.lua` does not do the value generation. Nor does it store the cache. The "value generation" is the actual Clickhouse query execution, and the cache storage happens in `set.lua`.

The get script
- Checks the cache
- Checks to see how many people are waiting on the wait queue
- If there is no one else besides the current query waiting, set a marker showing that the query is being executed and return
- Otherwise return the task timeout of the executing task

After this the "value generation" is run, which is a fancy word for the Clickhouse query. Then the `set.lua` script is run, which updates all the various keys.